### DISCUSSION TITLE
multi site & templates

### DISCUSSION THREAD

**Discussion by goshawk1980** [givanz/Vvveb#137] [General]
Been playing around since a couple of hours and got the multisite up and running correctly (well i think).
I noticed that editing the homepage while in multisite mode these changes are edited on both websites.

Is it correct you need to duplicate the template per website in order to prevent this from happening or am I missing something?
I assumed every subsite would have his own folder in the background where these changes would be saved. Leaving the template section itself clean, serving as a "template"

I also read somewhere in order to translate the homepage you need to duplicate the template as well.
If so this is wild :) when you manage 20 websites each in 5 different languages you will end up with hundred duplicates of the same template?? I must be doing something wrong cause that seems unrealistic ;)

**Comment by givanz** [comment]
>I noticed that when in multisite mode, you edit the homepage template these changes get's edited on both websites.
> Is it correct you need to duplicate the template per website in order to prevent this from happening or am I missing something?
> I assumed every subsite would have his own folder in the background where the changes would be saved.

Layout changes are saved in the theme html files, you need to duplicate a theme if you need to have different layouts on two sites.

> I also read somewhere in order to translate the homepage you need to duplicate it as a template.
> This is wild :) when you manage 20 websites each in 5 different languages you will end up with hundred duplicates of the same template?? I must be doing something wrong cause that seems unrealistic ;)

Content and translations are not saved in the theme, multilingual content is saved in the db and translations for theme text are saved in gettext files (.po/.mo) inside `/locale` folder.

Language packs can be installed from the language edit page for available packs.

Translations can be added/edited from admin > settings > localization > translations https://demo.vvveb.com/admin/index.php?module=localization/translations&action=domain&lang=ro_RO&domain=user

Translations are common for the entire setup and are available for all themes/sites.

If you need to translate a text from a theme don't change it from page builder but add the text in the translation list.

I will add a translate text dialog in the page builder as a shortcut to quickly translate template text in the next update with translation saved in gettext files.

You can also edit the translation files directly for faster translation from `locate/fr_FR/LC_MESSAGES/vvveb.po` with a text editor or something like Poedit.

If you want to share your translations you can push the changes to the language pack repository https://github.com/search?q=Vvveb%20language-pack&type=repositories

**Comment by goshawk1980** [comment]
Hi, 
Thanks for the fast support!

I'm unable to locate those .po files :)
When trying to add a translation i also get an error message right after hitting the save button ("Error saving! undefined")
Same happens on the demo site (link you provided). Anyways, I'll take a deeper look at the code tommorrow :) 

Sure, no problem in regards to the lang. rep.

**Comment by givanz** [comment]
Thanks for the bug report, it's fixed in the last commit https://github.com/givanz/Vvveb/commit/602a14ad3087cf245c21902d12a9a327524d9e47

You probably don't have a language pack installed, you can install a pack from the language edit page  
![language](https://github.com/givanz/Vvveb/assets/302738/df5aeb0b-273e-4931-b58f-71793e0de958)

Then the files are downloaded and you can edit them from `/locale/fr_FR/LC_MESSAGES/vvveb.po`

Or you can git clone the language pack repository in `/locale` folder.
